django/db/backends/sqlite3/schema.py

```diff
@@ -53,6 +55,63 @@ def quote_value(self, value):
+                raise NotSupportedError((
+                    'Renaming the "%s" table while in a transaction is not '
+                    'supported on SQLite because it would break references '
+                    'integrity. Try setting your `Migration.atomic` flag to '
+                    '`False` in order to allow the rename to take place'
+                ) % (old_db_table))
+            self.connection.enable_constraint_checking()
+            super().alter_db_table(model, old_db_table, new_db_table)
+            self.connection.disable_constraint_checking()
+        else:
+            super().alter_db_table(model, old_db_table, new_db_table)
+
+    def alter_field(self, model, old_field, new_field, strict=False):
+        old_field_name = old_field.name
+        if (new_field.name != old_field_name and
+                any(r.field_name == old_field.name for r in model._meta.related_objects)):
+            if self.connection.in_atomic_block:
+                raise NotSupportedError((
+                    'Renaming the "%s"."%s" column while in a transaction '
+                    'is not supported on SQLite because it would break '
+                    'references integrity. Try setting your `Migration.atomic` '
+                    'flag to `False` in order to the rename to take place.'
+                ) % (model._meta.db_table, old_field_name))
+            with atomic(self.connection.alias):
+                super().alter_field(model, old_field, new_field, strict=strict)
+                # Follow's SQLite documented procedure for performing changes
+                # that do no affect the on-disk content in any way.
+                # https://sqlite.org/lang_altertable.html#otheralter
+                with self.connection.cursor() as cursor:
+                    schema_version = cursor.execute('PRAGMA schema_version').fetchone()[0]
+                    cursor.execute('PRAGMA writable_schema = 1')
+                    table_name = model._meta.db_table
+                    references_template = ' REFERENCES "%s" ("%%s") ' % table_name
+                    old_column_name = old_field.get_attname_column()[1]
+                    new_column_name = new_field.get_attname_column()[1]
+                    search = references_template % old_column_name
+                    replacement = references_template % new_column_name
+                    cursor.execute('UPDATE sqlite_master SET sql = replace(sql, %s, %s)', (search, replacement))
+                    cursor.execute('PRAGMA schema_version = %d' % (schema_version + 1))
+                    cursor.execute('PRAGMA writable_schema = 0')
+                    # Raise an exception if the sqlite_master updates broke
+                    # anything to make sure the transaction is rolled back.
+                    if cursor.execute('PRAGMA integrity_check').fetchone()[0] != 'ok':
+                        raise Exception('Failed to repoint references from "%s"."%s" to "%s"."%s".' % (
```

Comment:
I think we might want to raise an `IntegrityError` with the output of `PRAGMA integrity_check` here instead. Not sure if `PRAGMA integrity_check` will raise an error by itself, you can test that out by changing `replacement` to invalid SQL.